### ISSUE TITLE
False positive color-contrast: `font-size: 0` elements should be ignored

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#4820]
This came up recently where an element with `font-size: 0` is throwing a color-contrast violation when the text is not visible. Normally an element with `font-size: 0` would have a width and height of 0 (which should be ignored), but explicitly setting the width and height of the element overrides that while still visually hidding the font.

```html
<ol>
  <li>Visible text</li>
  <li style="font-size: 0; width: 16px; height: 16px; background: red; color: #282828;">Not visible</li>
</ol>
```

**Comment by straker** [comment]
`font-size: 1px` seems to be used a lot in conjunction with whitespace characters (`&nbsp;`), in emails and tables for line clears (some of those use actual letters), and sometimes to hide text (mostly used in conjunction with setting the background to the page background). I've seen similar things with `font-size: 2px`.

However the more I look at it the more I wonder if we shouldn't try to guess the intent of the font-size use. I'd be fine with saying only `font-size: 0` should fail the matcher and let everything else through since font-size: 1px is technically still visible (albeit very tiny and needs a magnifying glass).

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core devbuild 

Not seeing any color-contrast issues when `font-size=0`

code below:

```
<ol>
  <li>Visible text</li>
  <li style="font-size: 0; width: 16px; height: 16px; background: red; color: #282828;">Not visible</li>
</ol>

```

https://github.com/user-attachments/assets/ba78e649-6c9d-4499-bef7-504444e6698b

## Environment:
| **Label**                        | **Value** |
| -------------------------- | ------ |
| **Product**               | _axe-core_ |
|**Version** | _4.10.3(Dev Branch Code base_ |
| **OS-Details**     | _MAC- M1-15.3 _ |
|**BrowserDetails**|_Chrome Version 138.0.7 (Official Build) (64-bit)_|
